DigitalLearningSolutions.Data.Tests/Services/CourseServiceTests.cs

```diff
@@ -546,6 +547,51 @@ public void GetCourseOptionAlphabeticalListForCentre_calls_data_service()
             result.Should().BeEquivalentTo(courseOptions);
         }
 
+        [Test]
+        public void GetApplicationOptionsAlphabeticalListForCentre_calls_data_service()
+        {
+            // Given
+            const int categoryId = 1;
+            const int centreId = 1;
+            var applicationOptions = new List<ApplicationDetails>();
+            A.CallTo(() => courseDataService.GetApplicationsAvailableToCentreByCategory(centreId, categoryId))
+                .Returns(applicationOptions);
+
+            // When
+            var result = courseService.GetApplicationOptionsAlphabeticalListForCentre(centreId, categoryId);
+
+            // Then
+            A.CallTo(() => courseDataService.GetApplicationsAvailableToCentreByCategory(centreId, categoryId))
+                .MustHaveHappenedOnceExactly();
+            result.Should().BeEquivalentTo(applicationOptions);
+        }
+
+        [Test]
+        public void GetApplicationOptionsAlphabeticalListForCentre_filters_by_topic_if_topic_given()
+        {
+            // Given
+            const int categoryId = 1;
+            const int centreId = 1;
+            const int topicId = 1;
+
+            var applicationOne = new ApplicationDetails { CourseTopicId = 1 };
+            var applicationTwo = new ApplicationDetails { CourseTopicId = 1 };
+            var applicationThree = new ApplicationDetails { CourseTopicId = 2 };
+
+            var applicationOptions = new List<ApplicationDetails>
+                { applicationOne, applicationTwo, applicationThree };
+            A.CallTo(() => courseDataService.GetApplicationsAvailableToCentreByCategory(centreId, categoryId))
+                .Returns(applicationOptions);
+
+            // When
+            var result = courseService.GetApplicationOptionsAlphabeticalListForCentre(centreId, categoryId, topicId)
+                .ToList();
+
+            // Then
+            result.All(s => s.CourseTopicId == 1).Should().BeTrue();

```

Comment:
This can also be written as `result.Should().OnlyContain(a => a.CourseTopicId == 1);`

Reply:
thanks, done